### ISSUE TITLE
Compiler: Consistent command expansion on all platforms?

### ISSUE THREAD

**Issue by mdwagner** [crystal-lang/crystal#12105]
## Discussion
Like I mentioned in my PR for the Interpreter (#12094), I noticed that Windows is expanding `ldflags` [in the compiler](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/compiler.cr#L346), but before it runs `Codegen (linking)`. 
However, I'm not sure why we wouldn't want to do the same thing [for other platforms](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/compiler.cr#L404) (i.e. Linux/Mac/Wasm)?
The only reason I could come up with, is that backticks are valid syntax when running a shell, but obviously Windows can't do it, so it was pulled out to run beforehand.
But I think that should've been the turning point to make it the same for all platforms.

My proposal would be to run the same command expansion that Windows is doing above, but for other platforms as well. This would make it more consistent between platforms and make the compiler and interpreter work more similarly.

Thoughts?

**Comment by straight-shoota** [comment]
The custom expansion proposed in https://github.com/crystal-lang/crystal/pull/12109 is quite limited. It only covers commands wrapped in backticks. And it's very simple. For example, there is no escaping. It also misses other features that a typical shell provides, most importantly variable expansion. So this might actually be a breaking change when an `ldflags` configuration depends on more advanced shell features.

So I'm skeptical about this approach, at least in a naive implementation.

Overall, we're missing a concept for managing linker flag configurations (see this forum thread for example: https://forum.crystal-lang.org/t/rfc-link-configuration/546/13).
And in this context I have some doubts whether expanding linker flags in a shell is actually a good idea.

**Comment by mdwagner** [comment]
Yeah, that makes sense. I'm not an expert in this kind of thing, just trying to provide some help for simple things, but it's clear I'm a little in-over-my-head here.
I think I'll close the linked PR for now.
